profile/c5g7-cmfd.cpp

```diff
@@ -0,0 +1,509 @@
+                        1, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 1,
+                        1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1};
+
+    std::map<int, Universe*> names = {{1, &mox43}, {2, &mox7}, {3, &mox87},
+                                      {4, &guide_tube}, {5, &fission_chamber}};
+    for(int n=0; n<17*17; n++)
+      matrix2[n] = names[mold[n]];  
+    
+    assembly2_lattice.setUniverses(17, 17, matrix2);
+  }
+  assembly2_cell.setFill(&assembly2_lattice);
+
+  /* Sliced up water cells - semi finely spaced */
+  Lattice refined_ref_lattice;
+  refined_ref_lattice.setWidth(0.126, 0.126);
+  Universe* refined_ref_matrix[10*10];
+  for(int n=0; n<10*10; n++)
+    refined_ref_matrix[n] = &reflector;
+  refined_ref_lattice.setUniverses(10, 10, refined_ref_matrix);
+  refined_reflector_cell.setFill(&refined_ref_lattice);
+
+  /* Sliced up water cells - right side of geometry */
+  Lattice right_ref_lattice;
+  right_ref_lattice.setWidth(1.26, 1.26);
+  Universe* right_ref_matrix[17*17];
+  for(int i=0; i<17; i++){
+    for(int j=0; j<17; j++){
+      int index =  17*j + i;
+      if(i<11)
+        right_ref_matrix[index] = &refined_reflector;
+      else
+        right_ref_matrix[index] = &reflector;
+    }
+  }
+  right_ref_lattice.setUniverses(17, 17, right_ref_matrix);
+  right_reflector_cell.setFill(&right_ref_lattice);
+
+  /* Sliced up water cells for bottom corner of geometry */
+  Lattice corner_ref_lattice;
+  corner_ref_lattice.setWidth(1.26, 1.26);
+  Universe* corner_ref_matrix[17*17];
+  for(int i=0; i<17; i++){
+    for(int j=0; j<17; j++){
+      int index = 17*j + i;
+      if(i<11 && j<11)
+        corner_ref_matrix[index] = &refined_reflector;
+      else
+        corner_ref_matrix[index] = &reflector;
+    }
+  }
+  corner_ref_lattice.setUniverses(17, 17, corner_ref_matrix);
+  corner_reflector_cell.setFill(&corner_ref_lattice);
+
+  /* Sliced up water cells for bottom of geometry */
+  Lattice bottom_ref_lattice;
+  bottom_ref_lattice.setWidth(1.26, 1.26);
+  Universe* bottom_ref_matrix[17*17];
+  for(int i=0; i<17; i++){
+    for(int j=0; j<17; j++){
+      int index = 17*j + i;
+      if(j<11)
+        bottom_ref_matrix[index] = &refined_reflector;
+      else
+        bottom_ref_matrix[index] = &reflector;
+    }
+  }
+  bottom_ref_lattice.setUniverses(17, 17, bottom_ref_matrix);
+  bottom_reflector_cell.setFill(&bottom_ref_lattice);
+
+  /* 4 x 4 core to represent two bundles and water */
+  Lattice full_geometry;
+  full_geometry.setWidth(21.42, 21.42);
+  Universe* universes[] = {
+    &assembly1,         &assembly2,         &right_reflector,
+    &assembly2,         &assembly1,         &right_reflector,
+    &bottom_reflector,  &bottom_reflector,  &corner_reflector}; 
+  full_geometry.setUniverses(3, 3, universes);
+  root_cell.setFill(&full_geometry);
+  
+  /* Create CMFD mesh */
+  log_printf(NORMAL, "Creating CMFD mesh...");
+
+  Cmfd cmfd;
+  cmfd.setMOCRelaxationFactor(0.6);
```

Comment:
What compiler are you using? I compile and run just fine with GNU on my machine.